gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java

```diff
@@ -211,9 +212,27 @@ private void handleRequest(final HttpServerRequest req, final Buffer bodyData, f
             cReq.headers().set("Authorization", "Basic " + base64UsernamePassword);
         }
 
-
         MultiMap headers = cReq.headers();
+        String hostHeaderBefore = HttpHeaderUtil.getHeaderValue(headers, HOST_HEADER);
         final HeaderFunctions.EvalScope evalScope = rule.getHeaderFunction().apply(headers);
+        String hostHeaderAfter = HttpHeaderUtil.getHeaderValue(headers, HOST_HEADER);
+        // see https://github.com/swisspush/gateleen/issues/394
+        if (hostHeaderAfter == null ||
```

Comment:
Somehow this block here doesn't make sense to me. Especially those points are unclear to me:

- Why do we already set the host header to our target value beforehand [a few lines above](https://github.com/swisspush/gateleen/blob/v1.2.2/gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java#L209)? So isn't our use-case already implemented by this line?
- IntelliJ inspector says that `hostHeaderAfter != null` is _"always true when reached"_ So do we really need that check?
- AFAIK this makes the headerFunctions useless for "Host" header as it strictly overrides the setting which the headerFunctions did. So why do we ever configure headerFunctions just to revert them in a hardcoded way? Why not just remove such (bad) headerFunctions in the first place from our config?

Reply:
The check is already optimized in the newest update. 
The HostHeader will only be set if there is a rule containing such a header config.  That's why we check this before and after the apply() to evaluate if it changed during this really. If it did change, we don't want to it to be overwritten by the default.

Reply:
BTW, the harcoded part is the old default which is magically working.  So the intention with this reduced set is "don't touch a running system". This means we do only apply the Host header for routes where this is explicitely configured. In all other cases the very old hardcoded default is still doing it's job.

Reply:
Ah now I see what you meant. This was a failed merge, I will correct this.